src/main/scala/datalog/storage/VolcanoStorageManager.scala

```diff
@@ -70,9 +93,19 @@ class VolcanoStorageManager(ns: NS = NS()) extends CollectionsStorageManager(ns)
                     if (r == d && !found && i > idx)
                       found = true
                       idx = i
-                      Scan(getKnownDeltaDB(r), r)
+                      withNegation(k.negated(i))(r, k.sizes(i),
```

Comment:
Again, it seems like calling withNegation even if there is no negation going on is unnecessary?